### PR TITLE
Add non-nix installation instructions

### PR DESCRIPTION
**Pull request by sveitser**
- Add instructions
- Run weekly CI job to execute installation instructions and run tests.

Room for improvement: currently the toolchain file is not read by nix so there is potentital for the toolchain file to go out of sync. It would be easy to do that however, in that case `nix flake update` won't update rust anymore. I read dependabot doesn't handle toolchain files so we'd have to manually update it. We could also use "stable" instead of a specific version tag.

### CODE REVIEW

**Comment by sveitser** [comment]
Since I removed the PR CI trigger, the latest run is [here](https://github.com/EspressoSystems/espresso-sequencer/actions/runs/6896439272/job/18762566284).

File: .github/workflows/ubuntu-install-without-nix.yml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+    steps:
+      - uses: actions/checkout@v4
+      - name: Install and test
+        # Execute all lines that start with four spaces, and remove leading 'sudo '
+        # because we're in a container and already root.
+        run: |
+          grep '^    ' doc/ubuntu.md \
+            | sed 's/^    //' \
+            | sed 's/^sudo //' \
+            | bash -exo pipefail
```
Comment (jbearer) [inline .github/workflows/ubuntu-install-without-nix.yml:27]: This is pretty nifty 👍

**Review by jbearer — APPROVED** [review]
LGTM

**Comment by nyospe** [comment]
Thank you for this.

**Pull request merged**